lib/segment/src/types.rs

```diff
@@ -1711,6 +1711,20 @@ impl PayloadSchemaType {
             Self::Uuid => PayloadSchemaParams::Uuid(UuidIndexParams::default()),
         }
     }
+
+    /// Check if this type supports a `match` condition
+    pub fn support_match(&self) -> bool {
+        match self {
+            Self::Keyword => true,
+            Self::Integer => true,
+            Self::Bool => true,
+            Self::Uuid => true,
+            Self::Float => false,
+            Self::Geo => false,
+            Self::Text => false,
+            Self::Datetime => false,
+        }
+    }
```

Comment:
integer and uuid can be parametrized without map index, which would make them not support match conditions

Reply:
I handled the integer the index here https://github.com/qdrant/qdrant/pull/6363/commits/d6fbc6d2fd0a83c22cc03187acb1abce91cd6055

But I do not think the UUID lookup can be disabled.